arcgis-ios-sdk-samples/Layers/ArcGIS vector tiled layer (custom style)/VectorStylesViewController.swift

```diff
@@ -21,31 +21,34 @@ protocol VectorStylesVCDelegate: AnyObject {
 }
 
 class VectorStylesViewController: UITableViewController {
+    /// The item IDs of the custom styles.
     var itemIDs: [String] = []
+    /// The item ID of the selected style.
     var selectedItemID: String?
     
     weak var delegate: VectorStylesVCDelegate?
     
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        if let selectedItemID = selectedItemID,
-            let row = itemIDs.firstIndex(of: selectedItemID) {
-            let indexPath = IndexPath(row: row, section: 0)
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
+        if itemIDs[indexPath.row] == selectedItemID {
+            // Select and indicate the displayed item.
+            cell.accessoryType = .checkmark
             tableView.selectRow(at: indexPath, animated: false, scrollPosition: .none)
```

Comment:
This seems odd to me. The "selected" row is both selected _and_ checked? It seems like it should be one or the other. Could this line be removed?

Reply:
`selectRow(at:animated:scrollPosition:)` doesn't prompt `tableView(_:didSelectRowAt:)` (ref: [Apple doc](https://developer.apple.com/documentation/uikit/uitableview/1614875-selectrow#discussion) ) so a checkmark won't display when the view is initially loaded. And if I don't select the row, `tableView(_:didDeselectRowAt:)`won't prompt appropriately. I could try to work around it but I think this is the cleanest solution. What do you think?

Reply:
Ah! I see what you mean. Right now, selection is actually being tracked in two places (sort of): 1) the table view's selected row (used to check/uncheck rows), and 2) `selectedItemID`, although currently that is only used to check a row when the table view is first displayed. Instead, I propose that `selectedItemID` should be used for both purposes, and that could easily be done by adding the following lines to `tableView(_:didSelectRowAt:)`:

```swift
if let selectedItemID = selectedItemID, let row = itemIDs.firstIndex(of: selectedItemID) {
    tableView.cellForRow(at: IndexPath(row: row, section: 0))?.accessoryType = .none
}
selectedItemID = itemID
```

Then the implementation of `tableView(_:didDeselectRowAt:)` isn't needed.

Reply:
Great idea! I cleaned up the code a bit and tried to make it easier to follow. Please let me know what you think!